lib/ansible/executor/play_iterator.py

```diff
@@ -142,6 +151,14 @@ class PlayIterator:
     FAILED_RESCUE = 4
     FAILED_ALWAYS = 8
 
+    VALID_FAILED_STATES = [
+        FAILED_NONE,
+        FAILED_SETUP,
+        FAILED_TASKS,
+        FAILED_RESCUE,
+        FAILED_ALWAYS,
+    ]
```

Comment:
I think with moving to 3.8 on the controller, we can explore using some more advanced python features. This seems like a fantastic candidate for using an [enum class type](https://docs.python.org/3/library/enum.html).